packages/core/contracts-ovm/insured-bridge/implementation/BridgePoolFactory.sol

```diff
@@ -0,0 +1,189 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity ^0.8.0;
+
+// Importing local copies of OVM contracts is a temporary fix until the @eth-optimism/contracts package exports 0.8.x
+// contracts. These contracts are relatively small and should have no problems porting from 0.7.x to 0.8.x, and
+// changing their version is preferable to changing this contract to 0.7.x and defining compatible interfaces for all
+// of the imported DVM contracts below.
+import "./OVM_CrossDomainEnabled.sol";
+import "./BridgePoolFactoryInterface.sol";
+import "../../../contracts/oracle/interfaces/IdentifierWhitelistInterface.sol";
+import "../../../contracts/oracle/interfaces/FinderInterface.sol";
+import "../../../contracts/oracle/implementation/Constants.sol";
+import "../../../contracts/common/interfaces/AddressWhitelistInterface.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+
+/**
+ * @notice Administrative contract deployed on L1 that has an implicit reference to a DepositBox. This contract is
+ * responsible for deploying new BridgePools, which houses passive liquidity and enables relaying of L2 deposits.
+ * @dev The owner of this contract can call permissioned functions on the L2 DepositBox.
+ */
+contract BridgePoolFactory is BridgePoolFactoryInterface, Ownable, OVM_CrossDomainEnabled {
+    // Finder used to point to latest OptimisticOracle and other DVM contracts.
+    address private finder;
+
+    // L2 Deposit contract that originates deposits that can be fulfilled by this contract.
+    address private depositContract;
+
+    mapping(address => L1TokenRelationships) private whitelistedTokens;
+
+    // Set upon construction and can be reset by Owner.
+    uint256 private optimisticOracleLiveness;
+    uint256 private proposerBondPct;
+    bytes32 private identifier;
+
+    event SetDepositContract(address indexed l2DepositContract);
+    event SetRelayIdentifier(bytes32 indexed identifier);
+    event SetOptimisticOracleLiveness(uint256 indexed liveness);
+    event SetProposerBondPct(uint256 indexed proposerBondPct);
+    event WhitelistToken(address indexed l1Token, address indexed l2Token, address indexed bridgePool);
+
+    constructor(
+        address _finder,
+        address _crossDomainMessenger,
+        uint256 _optimisticOracleLiveness,
+        uint256 _proposerBondPct,
+        bytes32 _identifier
+    ) OVM_CrossDomainEnabled(_crossDomainMessenger) {
+        finder = _finder;
+        require(address(_getCollateralWhitelist()) != address(0), "Invalid finder");
+        _setOptimisticOracleLiveness(_optimisticOracleLiveness);
+        _setProposerBondPct(_proposerBondPct);
+        _setIdentifier(_identifier);
+    }
+
+    // Admin functions
```

Comment:
nit: these are not consistent comments with your other contracts.

Reply:
Should I remove `Admin functions`? I added natspec

Reply:
to be the _most_ consistent perhaps wrap them in the `***` like we do in the other contracts and like you did in the `OVM_CrossDomainEnabled`